SimTracker/SiPhase2Digitizer/interface/Phase2TrackerDigitizerAlgorithm.h

```diff
@@ -0,0 +1,213 @@
+
+  const bool addThresholdSmearing;
+    
+  // pseudoRadDamage
+  const double pseudoRadDamage;       // Decrease the amount off freed charge that reaches the collector
+  const double pseudoRadDamageRadius; // Only apply pseudoRadDamage to pixels with radius<=pseudoRadDamageRadius
+
+  // The PDTable
+  // HepPDTable *particleTable;
+  // ParticleDataTable *particleTable;
+
+  //-- charge fluctuation
+  const double tMax;  // The delta production cut, should be as in OSCAR = 30keV
+
+  // The eloss fluctuation class from G4. Is the right place? 
+  const std::unique_ptr<SiG4UniversalFluctuation> fluctuate;   // make a pointer
+  const std::unique_ptr<GaussianTailNoiseGenerator> theNoiser;
+
+  //-- additional member functions    
+  // Private methods
+  void primary_ionization( const PSimHit& hit, std::vector<DigitizerUtility::EnergyDepositUnit>& ionization_points) const;
+  void drift(const PSimHit& hit,
+	     const Phase2TrackerGeomDetUnit* pixdet,
+	     const GlobalVector& bfield,
+	     const std::vector<DigitizerUtility::EnergyDepositUnit>& ionization_points,
+	     std::vector<DigitizerUtility::SignalPoint>& collection_points) const;
+  void induce_signal(const PSimHit& hit,
+		     const size_t hitIndex,
+		     const unsigned int tofBin,
+		     const Phase2TrackerGeomDetUnit* pixdet,
+		     const std::vector<DigitizerUtility::SignalPoint>& collection_points);
+  void fluctuateEloss(int particleId, float momentum, float eloss, 
+		      float length, int NumberOfSegments,
+		      float elossVector[]) const;
+  virtual void add_noise(const Phase2TrackerGeomDetUnit* pixdet, float thePixelThreshold);
+  virtual void pixel_inefficiency(const SubdetEfficiencies& eff,
+				  const Phase2TrackerGeomDetUnit* pixdet,
+				  const TrackerTopology* tTopo);
+  
+  virtual void pixel_inefficiency_db(uint32_t detID);
+
+  // access to the gain calibration payloads in the db. Only gets initialized if check_dead_pixels_ is set to true.
+  const std::unique_ptr<SiPixelGainCalibrationOfflineSimService> theSiPixelGainCalibrationService_;    
+  LocalVector DriftDirection(const Phase2TrackerGeomDetUnit* pixdet,
+			     const GlobalVector& bfield,
+			     const DetId& detId) const;
+  
+  virtual void module_killing_conf(uint32_t detID); // remove dead modules using the list in the configuration file PixelDigi_cfi.py
+  virtual void module_killing_DB(uint32_t detID);  // remove dead modules uisng the list in the DB
+
+  const SubdetEfficiencies subdetEfficiencies_;
+  
+  // For random numbers
+  const std::unique_ptr<CLHEP::RandFlat> flatDistribution_;
+  const std::unique_ptr<CLHEP::RandGaussQ> gaussDistribution_;
+  
+  // Threshold gaussian smearing:
+  const std::unique_ptr<CLHEP::RandGaussQ> smearedThreshold_Endcap_;
+  const std::unique_ptr<CLHEP::RandGaussQ> smearedThreshold_Barrel_;
+  
+  //for engine passed into the constructor from Digitizer
+  CLHEP::HepRandomEngine* rengine_;   
+
+  double calcQ(float x) const {
```

Comment:
@suchandradutta - I assume you are using this function instead of gsl one. Do you still need a dependency on gsl?